samples/net/latmon/src/main.c

```diff
@@ -0,0 +1,718 @@
+/*
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2024 Philippe Gerum <rpm@xenomai.org>
+ * Copyright (c) 2024 Jorge Ramirez-Ortiz <jorge.ramirez@oss.qualcomm.com>
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(latency_monitor, LOG_LEVEL_DBG);
+
+#include <stdio.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log_core.h>
+#include <zephyr/net_buf.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/net_core.h>
+#include <zephyr/net/net_context.h>
+#include <zephyr/net/net_mgmt.h>
+#include <zephyr/net/socket.h>
+#include <zephyr/sys/atomic.h>
+#include "latmon.h"
+
+#define	INADDR_BROADCAST	((uint32_t) 0xffffffff)
+
+/*
+ * Blink Control
+ * DHCP: red
+ * waiting for connection: blue
+ * sampling: green
+ */
+#define LED_WAIT_PERIOD 1000000
+#define LED_DHCP_PERIOD 500000
+#define LED_RUN_PERIOD  200000
+
+static const struct gpio_dt_spec *running_led;
+static const struct gpio_dt_spec led_run =
+	GPIO_DT_SPEC_GET(DT_ALIAS(led0), gpios);
+static const struct gpio_dt_spec led_wait =
+	GPIO_DT_SPEC_GET(DT_ALIAS(led1), gpios);
+static const struct gpio_dt_spec led_dhcp =
+	GPIO_DT_SPEC_GET(DT_ALIAS(led2), gpios);
```

Comment:
Not all board have 3 leds, maybe use `GPIO_DT_SPEC_GET_OR` and make them optional.

Reply:
ok